File: src/main/java/MoveZeroes/Axinghub/MoveZeros.java

/*
Runtime: 0 ms, faster than 100.00% of Java online submissions for Move Zeroes.
Memory Usage: 36.4 MB, less than 99.98% of Java online submissions for Move Zeroes.
 */

class Solution {
    public void moveZeroes(int[] nums) {
        int i=0,j=0;
        for(;i<nums.length;){
            if(nums[i]!=0){
                if(i!=j){
                    nums[j++]=nums[i++];
                }else{
                    j++;
                    i++;
                }
            }else{
                i++;
            }
        }
        for(;j<nums.length;){
            nums[j++]=0;
        }
    }
}

File: src/main/java/MoveZeroes/j-dean/moveZeros.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        for(int i=0; i<nums.size(); i++)
        {
            //当遇到0元素时，往后找到第一个非0的数与之互换
            if(nums[i] == 0)
            {
			    int j = i;
			    while(nums[j++] == 0 && j<nums.size());
			    nums[i] = nums[j-1];
			    nums[j-1] = 0;
            }
        }
    }
};
